proto/frequenz/api/dispatch/v1/dispatch.proto

```diff
@@ -326,19 +336,49 @@ message TargetComponents {
     repeated uint64 ids = 1;
   }
 
-  // Wrapper for controlling dispatches with a set of component categories
+  // Wrapper for controlling dispatches with a set of component categories and optional types
   // Required as we can't use `repeated` directly in a `oneof`
   message CategorySet {
     // Set of component categories
     repeated frequenz.api.common.v1.microgrid.components.ComponentCategory categories = 1;
   }
 
+  // Wrapper for controlling dispatches with a set of battery types
+  // Required as we can't use `repeated` directly in a `oneof`
+  message BatteryTypeSet {
+    // Set of component categories
+    repeated frequenz.api.common.v1.microgrid.components.BatteryType types = 1;
+  }
+
+  // Wrapper for controlling dispatches with a set of EV charger types
+  // Required as we can't use `repeated` directly in a `oneof`
+  message EvChargerTypeSet {
+    // Set of component categories
+    repeated frequenz.api.common.v1.microgrid.components.EvChargerType types = 1;
+  }
+
+  // Wrapper for controlling dispatches with a set of inverter types
+  // Required as we can't use `repeated` directly in a `oneof`
+  message InverterTypeSet {
+    // Set of component categories
+    repeated frequenz.api.common.v1.microgrid.components.InverterType types = 1;
+  }
+
   oneof components {
     // Set of component IDs
     IdSet component_ids = 1;
 
     // Component categories
     CategorySet component_categories = 2;
+
+    // Battery types
+    BatteryTypeSet batteries = 3;
+
+    // EV Charger types
+    EvChargerTypeSet ev_chargers = 5;
+
+    // Inverter types
+    InverterTypeSet inverters = 4;
```

Comment:
@llucax are these intended to be mutually exclusive? This would suggest you can't mix types and categories, so if a dispatch targets batteries, it can ONLY target batteries? (for example)